### ISSUE TITLE
ERROR: hotkey.lua:415: key must be a string or a number

### ISSUE THREAD

**Issue by ys64** [Hammerspoon/hammerspoon#3619]
I've been encountering this error for the past few days, and as a result, I am unable to use Hammerspoon. Is anyone else experiencing the same issue?

> 2024-03-17 14:24:29: *** ERROR: ...merspoon.app/Contents/Resources/extensions/hs/hotkey.lua:415: key must be a string or a number
stack traceback:
	[C]: in function 'error'
	...merspoon.app/Contents/Resources/extensions/hs/hotkey.lua:20: in upvalue 'getKeycode'
	...merspoon.app/Contents/Resources/extensions/hs/hotkey.lua:195: in function 'hs.libhotkey.new'
	...merspoon.app/Contents/Resources/extensions/hs/hotkey.lua:415: in function 'hs.libhotkey.bind'
	(...tail calls...)
	./core.fnl:143: in upvalue 'f'
	./lib/functional.fnl:155: in local 'f'
	./lib/functional.fnl:135: in function 'lib.functional.reduce'
	(...tail calls...)
	./core.fnl:145: in main chunk
	(...tail calls...)
	[C]: in function 'rawrequire'
	...poon.app/Contents/Resources/extensions/hs/_coresetup.lua:662: in function 'require'
	/Users/ys/.hammerspoon/init.lua:15: in main chunk
	[C]: in function 'xpcall'
	...poon.app/Contents/Resources/extensions/hs/_coresetup.lua:723: in function 'hs._coresetup.setup'
	(...tail calls...)

**Comment by muescha** [comment]
what is the code of your:

/Users/ys/.hammerspoon/init.lua:15: in main chunk

**Comment by ys64** [comment]
This is the code in line 15:
```
require "core"
```

**Comment by muescha** [comment]
Did you check if it starts with an empty `ini.lua`?

**Comment by ys64** [comment]
No, I have not. Actually, that is a great idea! I did not know I can run Hammerspoon with an empty `init.lua`.

Also, because of your comment, I was able to fix the issue.

The issue was caused by the following line:
```lua
package.path = package.path .. ";" .. "/opt/homebrew/opt/luarocks/share/lua/5.4/luarocks/?.lua"
```

I use `luarocks` and I used to have that line enabled to make it work.

However, commenting it out enables me to run Hammerspoon without any issues.

So, I am back to normal! Thank you very much! 😊

**Comment by muescha** [comment]
I am happy that my hint was helpful.

I also use luarocks, and I found this solution for me:

https://github.com/muescha/dot_hammerspoon/blob/master/Functions/SetupLuaRocks.lua

(I remember that this `luarocks` and `luasockets` was a requirement to enable debugging (with `mobdebug`) in IntelliJ IDEA)

**Comment by ys64** [comment]
Thank you for sharing your `SetupLuaRocks.lua`!

You are much more advanced than I am. I found a couple of interesting Functions in your repository! 😊
